packages/calcite-components/src/components/table/table.tsx

```diff
@@ -0,0 +1,510 @@
+ */
+
+@Component({
+  tag: "calcite-table",
+  styleUrl: "table.scss",
+  shadow: true,
+  assetsDirs: ["assets"],
+})
+export class Table implements LocalizedComponent, LoadableComponent, T9nComponent {
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /** Specifies the appearance of the component. */
+  @Prop({ reflect: true }) appearance: TableAppearance = "simple";
+
+  /** Specifies an accessible title for the component. */
+  @Prop() caption!: string;
+
+  /** When `true`, number values are displayed with a group separator corresponding to the language and country format. */
+  @Prop({ reflect: true }) groupSeparator = false;
+
+  /** Specifies the layout of the component. */
+  @Prop({ reflect: true }) layout: TableLayout = "auto";
+
+  /** When `true`, displays the position of the row in numeric form */
+  @Prop({ reflect: true }) numbered = false;
+
+  /** Specifies the Unicode numeral system used by the component for localization. */
+  @Prop({ reflect: true }) numberingSystem?: NumberingSystem;
+
+  /** Specifies the page size of the component. When true, renders pagination. */
+  @Prop({ reflect: true }) pageSize = 0;
+
+  /** Specifies the size of the component. */
+  @Prop({ reflect: true }) scale: Scale = "m";
+
+  /** Specifies the selection mode of the component. */
+  @Prop({ reflect: true }) selectionMode: Extract<"none" | "multiple" | "single", SelectionMode> =
+    "none";
+
+  @Watch("groupSeparator")
+  @Watch("numbered")
+  @Watch("numberingSystem")
+  @Watch("pageSize")
+  @Watch("scale")
+  @Watch("selectionMode")
+  handleNumberedChange(): void {
+    this.updateRows();
+  }
+
+  /**
+   * Specifies the component's selected items.
+   *
+   * @readonly
+   */
+  @Prop({ mutable: true }) selectedItems: HTMLCalciteTableRowElement[] = [];
+
+  /**
+   * Made into a prop for testing purposes only
+   *
+   * @internal
+   */
+  // eslint-disable-next-line @stencil-community/strict-mutable -- updated by t9n module
+  @Prop({ mutable: true }) messages: TableMessages;
+
+  /**
+   * Use this property to override individual strings used by the component.
+   */
+  // eslint-disable-next-line @stencil-community/strict-mutable -- updated by t9n module
+  @Prop({ mutable: true }) messageOverrides: Partial<TableMessages>;
+
+  @Watch("messageOverrides")
+  onMessagesChange(): void {
+    /* wired up by t9n util */
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Element() el: HTMLCalciteTableElement;
+
+  @State() colCount = 0;
+
+  // this could probably also be a public property for setting the currently active page on load
```

Comment:
Agreed

Reply:
Will be follow-up.

Reply:
Added to follow-up issue: https://github.com/Esri/calcite-design-system/issues/7630